incomplete-envelopes-alert.tf

```diff
@@ -0,0 +1,31 @@
+module "bulk-scan-incomplete-envelopes-alert" {
+  source            = "git@github.com:hmcts/cnp-module-metric-alert"
+  location          = "${azurerm_application_insights.appinsights.location}"
+  app_insights_name = "${azurerm_application_insights.appinsights.name}"
+
+  enabled    = "${var.env == "prod"}"
+  alert_name = "Bulk_Scan_incomplete_envelopes_-_BSP"
+  alert_desc = "Triggers when bulk scan processor receives a log entry about having incomplete envelopes greater than 0"
+
+  app_insights_query = <<EOF
+traces
+| where severityLevel == 2
+| extend logger = tostring(customDimensions.LoggerName)
+| where logger endswith "IncompleteEnvelopesTask"
+| extend envelope_count = toint(extract("^There are (\\d+)", 1, message))
+| extend event_date = extract("^There are \\d+ incomplete envelopes as of ([0-9\\-]{10})$", 1, message)
+| where envelope_count > 0
```

Comment:
we don't log a warning when there are 0 incomplete envelopes so this can be ommited

Reply:
as a stand-alone query made with logical step with extra filter on

similarly like read some time ago elsewhere: "just to be sure".

performance-wise it's not exhausting action